### PR TITLE
✨ add flag to match the ds and dbGaP on aliquot id

### PR DESCRIPTION
**Pull request by chris-s-friedman**
adds a flag to match on aliquot id (instead of sample id). defaults on matching on sample ID.

### CODE REVIEW

**Comment by chris-s-friedman** [comment]
Do we want to update the readme with the new `--match_aliquot` flag?

**Comment by fiendish** [comment]
Yes please

**Comment by fiendish** [comment]
Do you know if just switching from sample to aliquot entirely would break any studies?

**Comment by chris-s-friedman** [comment]
I imagine it certainly could. but I'd have to do some investigating.

File: main.py [hunk 1]
```diff
@@ -36,11 +36,20 @@
...
+            "Match dbGaP `SAMPLE_ID` to `external_aliquot_id` in the dataservice.\n"  # noqa E501
+            " - Defaults to match on `external_sample_id`"
+        ),
+    )
     args = parser.parse_args()
     print(f"Args: {args.__dict__}")
 
     patches, alerts = ConsentProcessor(
-        args.server, args.db_url
+        args.server, args.db_url, args.match_aliquot
```
Comment (fiendish) [inline main.py:52]: I think this argument should go into get_patches_for_study since it seems like a study-specific issue?

Comment (fiendish) [inline main.py:52]: I think this argument should go into get_patches_for_study since it seems like a study-specific issue?

**Comment by fiendish** [comment]
Thanks! Please squash and rebase on master and I'll merge it in.

**Comment by fiendish** [comment]
Thank you!

**Pull request merged**